app/modules/intelligence/provider/provider_service.py

```
@@ -78,256 +85,157 @@ async def set_global_ai_provider(self, user_id: str, provider: str):
+                **common_params
+            }
+            
+            if not os.getenv("isDevelopmentMode") == "enabled":
+                model_params.update({
+                    "base_url": PORTKEY_GATEWAY_URL,
+                    "default_headers": portkey_headers,
+                })
+
+            return model_class(**model_params)
 
+    @lru_cache(maxsize=3)
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)